platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -0,0 +1,826 @@
+
+Dependencies on the TSS-Roster
+
+* At minimum, the `Roster` API is defined with an AddressBook wrapper for initial development and testing.
+* Delivery of TSS-Ledger-ID will require that the TSS-Roster proposal is delivered in a prior release.
+
+Impacts of TSS-Ledger-ID to Services Team
+
+* The timing of submitting the candidate roster may need to be adjusted to allow enough time for the candidate
+  roster to have key material generated.
+* Services will need to know or detect when the candidate roster fails to be adopted due to not having enough key
+  material generated.
+* The node's long term EC public key will need to be added to the address book.
+    * This requires modification of the HAPI transactions that manage the address book.
+
+Impacts to DevOps Team
+
+* Each consensus node will need a new long term EC key in addition to the existing RSA key.
+    * EC Key generation will have to happen before a node can join the network.
+* A new node added to an existing network will need to be given a state from an existing node after the network has
+  adopted the address book containing the new node.
+
+Implications Of Completion
+
+* The consensus nodes will be able to create ledger signatures for use in block proofs.
+* TSS-Block-Signing proposal becomes unblocked for development.
+
+### Requirements
+
+TSS Core Requirements
+
+| Requirement ID | Requirement Description                                                                                                      |
+|----------------|------------------------------------------------------------------------------------------------------------------------------|
+| TSS-001        | The ledger MUST have a public/private signing key pair that does not change when the consensus roster changes.               |
+| TSS-002        | The ledger private key MUST be a secret that nobody knows.                                                                   |
+| TSS-003        | The `ledger id` MUST be a public key that everyone knows and able to verify signatures from the ledger private key.          |
+| TSS-004        | The ledger signature for a message MUST be produced through a threshold number of consensus nodes signing the message.       |
+| TSS-005        | There MUST be a way to bootstrap an existing network's consensus roster with to give the ledger a public/private signing key |
+| TSS-006        | There MUST be a way for a new network to generate a public/private signing key for the ledger.                               |
+| TSS-007        | There MUST be a way to rotate the ledger's public/private signing key.                                                       |
+| TSS-008        | The ledger signature SHOULD be verifiable by an EVM smart contract without excessive cost or time.                           |
+| TSS-009        | The TSS implementation SHOULD be able to switch elliptic curves.                                                             |
```

Comment:
I stated this before, but i'll repeat - some of these requirements apply to the TSS-Library proposal instead of this one

Reply:
It's the requirements at this level that determined which TSS approach we went with.  If there was an existing library and we didn't have to write our own implementation, this requirement would still exist and it would get documented here in determining which library we choose.